src/main/core/worker.c

```diff
@@ -501,7 +501,11 @@ gboolean worker_scheduleTaskAtEmulatedTime(TaskRef* task, Host* host, EmulatedTi
         return FALSE;
     }
 
-    Event* event = event_new_(task, emutime_sub_emutime(t, EMUTIME_SIMULATION_START), host, host);
+    GQuark hostID = host_getID(host);
+    guint64 eventID = host_getNewEventID(host);
```

Comment:
It's a little odd for this to be the caller's responsibility. Maybe `event_new_` could still take the Host pointers, or at least the source host pointer, and just not store it.

Reply:
I agree it's odd, but it's mainly because the "event ID" itself is odd (it only exists for determinism/ordering purposes, and we could remove it by replacing it with an internal atomic global counter instead). But I'll change `event_new_` to take `..., Host* srcHost, GQuark dstHostID` instead.